runtime/ptrace_check.cc

```diff
@@ -114,7 +115,65 @@ static bool EnableTracingManually(const Settings& vm_settings) {
   return true;
 }
 
+static bool IsTracingCheckUnnecessaryOnVersion() {
+#if !OS_IOS || TARGET_OS_SIMULATOR
+  return true;
+#endif
+
+  // Check for arm64e.
+  cpu_type_t cputype = 0;
+  size_t cputype_size = sizeof(cpu_type_t);
+  if (::sysctlbyname("hw.cputype", &cputype, &cputype_size, nullptr, 0) < 0) {
+    FML_LOG(ERROR) << "Could not execute sysctl() to get CPU type: "
+                   << strerror(errno);
+  }
+
+  cpu_subtype_t cpusubtype = 0;
+  if (::sysctlbyname("hw.cpusubtype", &cpusubtype, &cputype_size, nullptr, 0) <
+      0) {
+    FML_LOG(ERROR) << "Could not execute sysctl() to get CPU subtype: "
+                   << strerror(errno);
+  }
+
+  // Tracing is necessary unless the device is arm64e (A12 chip or higher).
+  if (cputype != CPU_TYPE_ARM64 || cpusubtype != CPU_SUBTYPE_ARM64E) {
+    return false;
+  }
+
+  // Check for iOS 14.2 and higher.
```

Comment:
up to you but it feels like we're building in a lot of platform specific knowledge at this point. How about just letting this file statically take a SetOSVersionMayNeedTracingCheck injection or something and FlutterEngine can just set it to https://developer.apple.com/documentation/foundation/nsprocessinfo/1414876-isoperatingsystematleastversion? Could save us some work.

Reply:
If this were all in obj-c++ world then I wouldn't have bothered with all these syctl checks.  But it's being injected from:
https://github.com/flutter/engine/blob/e78e405e6a2fcb79956280be019e916a12a8de05/runtime/dart_vm.cc#L333

Reply:
that's what I meant injecting it statically (since all these stuff is static anyway). Since the engine calls first, we know it would be set before dart calls it again.

Reply:
I don't love that the lower level code would require a static variable to be set before it can be called, without enforcing that ordering itself.  Seems like an easy opportunity for bugs if this ever gets refactored.  Will defer to @chinmaygarde's opinion here.

If we decide to keep this `kern.osversion` parsing I'll add a unit test.

Reply:
I love the initiative in doing this in C, but let's make this an ObjC++ TU and use `NSOperatingSystemVersion` instead. You can remove the `TRACING_CHECKS_NECESSARY` macro and add a stub C++ file for non-Darwin platforms.